security/src/main/java/io/airlift/security/utils/KerberosUtil.java

```diff
@@ -0,0 +1,243 @@
+                        GSSCredential serverCredential = gssManager.createCredential(
+                                gssServiceName,
+                                GSSCredential.INDEFINITE_LIFETIME,
+                                SUPPORTED_OIDS,
+                                GSSCredential.ACCEPT_ONLY);
+
+                        context = gssManager.createContext(serverCredential);
+                    }
+                    context.requestMutualAuth(true);
+                    context.requestConf(true);
+                    context.requestInteg(true);
+                    context.requestCredDeleg(false);
+
+                }
+                catch (GSSException e) {
+                    throw Throwables.propagate(e);
+                }
+                return context;
+            }
+        });
+        return gssContext;
+    }
+
+    public static class KerberosConfiguration
+            extends Configuration
+    {
+        public static final String GSS_SERVER = "gss-server";
+        public static final String GSS_CLIENT = "gss-client";
+        private final String principal;
+        private final String krb5Conf;
+        private static final Logger log = Logger.get(KerberosConfiguration.class);
+
+        public KerberosConfiguration(String subjectName, String krb5Config, boolean fromClient)
+                throws UnknownHostException
+        {
+            if (fromClient) {
+                this.principal = subjectName;
+            }
+            else {
+                this.principal = KerberosUtil.getServicePrincipal(subjectName, null);
+            }
+            this.krb5Conf = krb5Config;
+        }
+
+        @Override
+        public AppConfigurationEntry[] getAppConfigurationEntry(String name)
+        {
+            System.setProperty("java.security.krb5.conf", krb5Conf);
+            String ticketCache = System.getenv("KRB5CCNAME");
+
+            Map<String, String> options = new HashMap<String, String>();
+            options.put("refreshKrb5Config", "true");
+            options.put("doNotPrompt", "true");
+            if (log.isDebugEnabled()) {
+                options.put("debug", "true");
+            }
+            switch (name) {
+                case GSS_SERVER:
+                    options.put("isInitiator", "false");
+                    options.put("useKeyTab", "true");
+                    if (principal != null) {
+                        options.put("principal", principal);
+                    }
+                    options.put("storeKey", "true");
+                    break;
+                case GSS_CLIENT:
+                    if (ticketCache != null) {
+                        options.put("ticketCache", ticketCache);
+                        options.put("useTicketCache", "true");
+                        options.put("renewTGT", "true");
+                    }
+                    break;
+                default:
+                    checkState(name == GSS_SERVER || name == GSS_CLIENT,
```

Comment:
How can this ever be true? Just throw an exception with the message below. BTW, it should be an IllegalArgumentException, not IllegalStateException.